### PR TITLE
Added option for exceptions

### PR DESCRIPTION
**Pull request by magnuswatn**
Hi,

I added the option for exceptions. It's simple and the GUI is rudimentary, but it works :-)

### CODE REVIEW

**Comment by lgarron** [comment]
Thanks for implementing!

I haven't tried it yet, but the basic idea sounds fine. I'll leave comments on the code.

File: content.js [hunk 1]
```diff
@@ -17,14 +17,18 @@ function defaultTheme() {
 
 if (!window.isSecureContext) {
   chrome.storage.local.get({
-    theme: "default"
+    theme: "default",
+    exceptions: ""
   }, function(items) {
     var theme = items.theme;
+    var exceptions = items.exceptions.split(',')
```
Comment (lgarron) [inline content.js]: However, I'm a little concerned about parsing an exception list on every page load. I don't think you can store `Set`s in `chrome.storage`, but could you store it as a map?

    {
      "bing.com": true
      "example.com": true
    }

File: content.js [hunk 2]
```diff
@@ -17,14 +17,18 @@ function defaultTheme() {
...
+        .map(Function.prototype.call, String.prototype.trim);
     if (themes.indexOf(theme) === -1) {
       theme = defaultTheme();
     }
-
-    document.body.classList.add("ugly-http-theme-" + theme);
+    if (!exceptions.includes(window.location.host)) {
```
Comment (lgarron) [inline content.js]: `host` can include an optional port. Could you use `hostname` instead?

File: options/options.html [hunk 3]
```diff
@@ -10,28 +10,53 @@
...
+        </select>
+        </td>
+      </tr>
+      <tr />
+      <tr>
+        <td>
+          Exceptions:
+        </td>
+        <td>
+          <input type="text" id="exceptions">
```
Comment (lgarron) [inline options/options.html]: Could we make this a `textarea` with newlines between hosts?

**Comment by magnuswatn** [comment]
Thanks for the feedback! I will have a look at this again tomorrow.

**Comment by magnuswatn** [comment]
Sorry this took so long, time flew by... I had a go again, it would be great if you could take a look!

File: options/options.js [hunk 4]
```diff
@@ -10,20 +12,45 @@ function display_status(message) {
...
 function restore_options() {
   chrome.storage.local.get({
-    theme: "default"
+    theme: "default",
+    exceptions: ""
   }, function(items) {
     document.getElementById("theme").value = items.theme;
+    var exceptionList = document.getElementById("exceptions");
+    for (var exception in items.exceptions) {
+        exceptionList.value += exception + '\n';
```
Comment (lgarron) [inline options/options.js:48]: Does the trailing newline cause an empty entry when serializing, or is that avoided?

File: options/options.js [hunk 5]
```diff
@@ -10,20 +12,45 @@ function display_status(message) {
...
+  timeout = window.setTimeout(save_options, 1000);
+}
+
 // Restores select box and checkbox state using the preferences
 // stored in chrome.storage.
 function restore_options() {
   chrome.storage.local.get({
-    theme: "default"
+    theme: "default",
+    exceptions: ""
```
Comment (lgarron) [inline options/options.js]: (same as above)

File: content.js [hunk 6]
```diff
@@ -17,14 +17,17 @@ function defaultTheme() {
 
 if (!window.isSecureContext) {
   chrome.storage.local.get({
-    theme: "default"
+    theme: "default",
+    exceptions: ""
```
Comment (lgarron) [inline content.js]: This should be `{}` now, right?

**Review by lgarron — CHANGES_REQUESTED** [review]
Thanks! I've left a few more comments about details.

File: content.js (hunk 6 above)
Comment (magnuswatn) [reply content.js]: You are, of course, absolutely correct. My bad. I have fixed this now.

File: options/options.js (hunk 4 above)
Comment (magnuswatn) [reply options/options.js:48]: That is avoided by:
`if (item)` (on line 20)
An empty string will be false, and thus is not added.

**Comment by lgarron** [comment]
Merged and released. Thanks!

**Pull request merged**